cloudformation/media-atom-maker.json

```diff
@@ -396,6 +433,52 @@
             }
         },
 
+        "UploaderS3EventsPermission": {
+          "Type": "AWS::Lambda::Permission",
+          "Properties": {
+            "Action": "lambda:InvokeFunction",
+            "FunctionName": { "Ref": "UploaderS3EventsLambda" },
+            "Principal": "s3.amazonaws.com",
+            "SourceArn": { "Fn::Join": ["", [ "arn:aws:s3:::", { "Fn::FindInMap": [ "StageMap", {"Ref": "Stage"}, "UploadBucket"] } ]] }
+          }
+        },
+
+        "UploaderDynamoEventsLambda": {
+          "Type": "AWS::Lambda::Function",
+          "Properties": {
+            "FunctionName":  {"Fn::Join": ["-", ["media-atom-maker-uploader-dynamo-events", {"Ref": "Stage"} ] ] },
+            "Description": "Triggers upload actions based on Dynamo events",
+            "Handler": "com.gu.media.DynamoEventsLambda::handleRequest",
+            "Code": {
+              "S3Bucket": { "Ref": "BuildBucket" },
+              "S3Key": { "Fn::FindInMap": [ "LambdaBuilds", "Uploader", { "Ref": "Stage" }] }
+            },
+            "Environment": {
+              "Variables": {
+                "REGION": { "Ref": "AWS::Region" },
+                "STACK": { "Ref": "Stack" },
+                "APP": "media-atom-uploader",
+                "STAGE": { "Ref": "Stage" },
+                "CONFIG_BUCKET": { "Ref": "ConfigBucket" },
+                "CONFIG_KEY": { "Fn::FindInMap": [ "LambdaConfig", "Uploader", { "Ref": "Stage" }] }
+              }
+            },
+            "MemorySize": 256,
+            "Role":  { "Fn::GetAtt": [ "UploaderRole", "Arn" ] },
+            "Runtime": "java8",
+            "Timeout": 300
+          }
+        },
+
+        "UploaderDynamoEventsMapping": {
```

Comment:
Connects the lambda to a stream of updates to the tracking table. It will then use that to make decisions (upload a part to YouTube etc)